File: src/main/java/chat/teco/tecochat/chat/domain/chat/event/ChatCreatedEventHistory.java
package chat.teco.tecochat.chat.domain.chat.event;

import chat.teco.tecochat.common.event.BaseEventHistory;
import jakarta.persistence.DiscriminatorValue;
import jakarta.persistence.Entity;
import java.time.LocalDateTime;

@Entity
@DiscriminatorValue("ChatCreatedEventHistory")
public class ChatCreatedEventHistory extends BaseEventHistory {

    private Long chatId;

    protected ChatCreatedEventHistory() {
    }

    public ChatCreatedEventHistory(
            LocalDateTime eventDateTime,
            Long chatId
    ) {
        super(eventDateTime);
        this.chatId = chatId;
    }

    public Long chatId() {
        return chatId;
    }
}

File: src/main/kotlin/chat/teco/tecochat/domain/chat/ChatCreatedEventHistory.kt
package chat.teco.tecochat.domain.chat

import chat.teco.tecochat.support.domain.BaseEventHistory
import jakarta.persistence.DiscriminatorValue
import jakarta.persistence.Entity
import java.time.LocalDateTime

@Entity
@DiscriminatorValue("ChatCreatedEventHistory")
class ChatCreatedEventHistory(
    override var eventDateTime: LocalDateTime,
    var chatId: Long,
) : BaseEventHistory(eventDateTime)
